bson/bsoncodec/map_codec.go

```diff
@@ -204,3 +199,80 @@ func clearMap(m reflect.Value) {
 		m.SetMapIndex(k, none)
 	}
 }
+
+func (mc *MapCodec) encodeKey(val reflect.Value) (string, error) {
+	if mc.MgoKeyHandling {
+		return fmt.Sprint(val), nil
+	}
+
+	// keys of any string type are used directly
+	if val.Kind() == reflect.String {
+		return val.String(), nil
```

Comment:
This has the same (or very similar) behaviour to `fmt.Sprint(val)` above - i.e. both use the stringer? Was this intended to be `string(val)` instead to use it directly rather than calling the `String`er?